azure/functions_worker/bindings/generic.py

```diff
@@ -0,0 +1,78 @@
+import typing
+
+from . import meta
+from .. import protos
+
+
+class GenericConverter(meta.InConverter,
+                       meta.OutConverter,
+                       binding='generic'):
+
+    @classmethod
+    def check_input_type_annotation(
+            cls, pytype: type, datatype: protos.BindingInfo.DataType) -> bool:
+        if (datatype is protos.BindingInfo.binary):
+            return issubclass(pytype, bytes)
+        elif (datatype is protos.BindingInfo.string):
+            return issubclass(pytype, str)
+        else:  # Unknown datatype
+            return False
```

Comment:
Handling conversion to `Plain old Python Object` if the function signature specifies a python object type via annotation, will unblock adding new bindings that use direct types without having to create converters.  Here is an example of SingnalR binding in Java:https://github.com/Azure/azure-functions-signalrservice-extension/blob/dev/binding-library/java/src/main/java/com/microsoft/azure/functions/signalr/SignalRConnectionInfo.java , this is a POJO

Reply:
There is no such thing as a "plain old Python object", unfortunately.  `dataclasses` in Python 3.7 is the closest thing, and IMO support for that should be a separate issue.